mysql/check.py

```diff
@@ -1008,24 +1014,53 @@ def _get_stats_from_innodb_status(self, db):
                 results['Innodb_os_file_writes'] = long(row[4])
                 results['Innodb_os_file_fsyncs'] = long(row[8])
             elif line.find('Pending normal aio reads:') == 0:
-                # Pending normal aio reads: 0, aio writes: 0,
-                # or Pending normal aio reads: 0 [0, 0] , aio writes: 0 [0, 0] ,
-                # or Pending normal aio reads: [0, 0, 0, 0] , aio writes: [0, 0, 0, 0] ,
-                # or Pending normal aio reads: 0 [0, 0, 0, 0] , aio writes: 0 [0, 0, 0, 0] ,
-                if len(row) == 14:
-                    results['Innodb_pending_normal_aio_reads'] = long(row[4])
-                    results['Innodb_pending_normal_aio_writes'] = long(row[10])
+                if len(row) == 8:
+                    # (len(row) == 8)  Pending normal aio reads: 0, aio writes: 0,
+                    if row[4].isdigit() and row[7].isdigit():
```

Comment:
@nmuesch Originally did this as you suggest; but on only one `if`.  I suggested this as a way to catch new formats going forward.  Could also wrap the entire section in `try...except` i suppose.